src/coredump/coredumpctl.c

```diff
@@ -992,8 +992,14 @@ static int save_core(sd_journal *j, FILE *file, char **path, bool *unlink_temp)
                         return r;
                 assert(r > 0);
 
+                /* If a journal directory or file is passed we might be looking for a core file on the host,
+                 * for example when accessing a core generated by nspawn, so try again without arg_root. */
                 r = chase_and_access(filename, arg_root, CHASE_PREFIX_ROOT, F_OK, &resolved);
-                if (r < 0)
+                if (r == -ENOENT && (arg_directory || arg_file)) {
```

Comment:
I'm not sure why we need to hide this behind `arg_directory` or `arg_file`? The container's journal could also go directory to the host journal no?

Reply:
Sure, but I'm thinking about nspawn's `--link-journal=host` switch. If that's used, the coredump journal entry will be located in the host's journal and --directory or --file shouldn't be needed in order to access that journal entry, so I'd drop the check for arg_directory or arg_file here and do it unconditionally.

Instead, we should always search both the container's journal and the host's journal and always try to read the machine ID of the root we're operating on (including the host) and add an extra `_MACHINE_ID=` journal match to make sure we only process coredumps belonging to the root we're operating on.

Reply:
Why wouldn't it? All the journal entries related to the image will have `_MACHINE_ID=<image-machine-id>` set. So if we read the machine ID from the image, we'll only get journal entries related to the image, regardless of whether they're stored in the image's journal or in the host's journal.